From 5c903bcc42fd5b053bb6ceee0bf3331a417538cb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Mon, 24 Jul 2017 13:11:33 +0100
Subject: [PATCH] cairo_canvas: shape clipping box doesn't match where its text
 is rendered

since (I believe)

commit 34d7602954d4483b3bc9db700e7df2c15348947a
Date:   Wed Nov 9 15:22:43 2016 +0200

    tdf#55469 Consistent line spacing across platforms

the point of that was to have the same line spacing on all platforms
to fix the cairo text layout bounds to use the same algorithm as
the generic text layout

Change-Id: I26d3dec8354a9eac1423557f7d52a08f37c8843c
---
 canvas/source/cairo/cairo_textlayout.cxx | 2 +-
 canvas/source/vcl/textlayout.cxx         | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/canvas/source/cairo/cairo_textlayout.cxx b/canvas/source/cairo/cairo_textlayout.cxx
index e0e9105..bb6b38d 100644
--- a/canvas/source/cairo/cairo_textlayout.cxx
+++ b/canvas/source/cairo/cairo_textlayout.cxx
@@ -162,7 +162,7 @@ namespace cairocanvas
 
         setupLayoutMode( *pVDev.get(), mnTextDirection );
 
-        const sal_Int32 nAboveBaseline( -aMetric.GetInternalLeading() - aMetric.GetAscent() );
+        const sal_Int32 nAboveBaseline( -aMetric.GetAscent() );
         const sal_Int32 nBelowBaseline( aMetric.GetDescent() );
 
         if( maLogicalAdvancements.getLength() )
diff --git a/canvas/source/vcl/textlayout.cxx b/canvas/source/vcl/textlayout.cxx
index b6fcd5f..3b86219 100644
--- a/canvas/source/vcl/textlayout.cxx
+++ b/canvas/source/vcl/textlayout.cxx
@@ -239,7 +239,7 @@ namespace vclcanvas
 
         setupLayoutMode( *pVDev.get(), mnTextDirection );
 
-        const sal_Int32 nAboveBaseline( /*-aMetric.GetIntLeading()*/ - aMetric.GetAscent() );
+        const sal_Int32 nAboveBaseline( -aMetric.GetAscent() );
         const sal_Int32 nBelowBaseline( aMetric.GetDescent() );
 
         if( maLogicalAdvancements.getLength() )
-- 
2.9.4

